### ISSUE TITLE
If Condition, nolock, declare may be not working on transactsql

### ISSUE THREAD

**Issue by mdora7** [taozhi8833998/node-sql-parser#1094]
**Describe the bug**
If condition for control somethings, nolock with some tables, also declare on head of select statement was not parsed.

**Database Engine**
MSSQL(T-SQL)

**To Reproduce**
--- =============================================
-- Author	   : mdora
-- MDate : 09.08.17
-- =============================================
declare @invoicenumber varchar(16)

set @invoicenumber = ''

if(len(@invoicenumber) < 16)
	set @invoicenumber = @invoicenumber + '%'

SELECT
	  '' AS InetSalePaymentBroker,
	   null AS InetSalePaymentDate,      
      '' AS SenderMersisNo,
	  '' AS SenderIsletmeMerkezi,
	  '' AS SenderTicaretSicilNo,
NULL AS isExport,
	  NULL AS IhracKayitKodu

	  FROM LG_001_01_INVOICE INV
      LEFT OUTER JOIN LG_001_CLCARD AS CLIENT ON (CLIENT.LOGICALREF = INV.CLIENTREF)
	  LEFT OUTER JOIN LG_001_01_STFICHE AS STFICHE (nolock)  ON (INV.FICHENO = STFICHE.INVNO)
      WHERE
      INV.TRCODE IN (3,6,7,8,9, 13)
      AND INV.DATE_ BETWEEN Convert(datetime,'',102) AND Convert(datetime,'',102)
 
 AND (
 
                       (@invoicenumber like '$%')
                      OR
                       (INV.FICHENO like @invoicenumber)
		  )
-4.4.0
-14.17.5

**Comment by goldmanpark** [comment]
So far, I have found following malfunctions:
1. TOP (n)
2. with(nolock)
3. table valued function at from clause
4. ServerName.DatabaseName.DBOwner.TableName: SQL Server structure
5. large bracket for column, table
I tried to fork, edit and PR but failed to debug it in console.

**Comment by taozhi8833998** [comment]
> Im also facing the same issue with TOP (n)

try the latest version `4.9.0`

**Comment by kasvith** [comment]
@taozhi8833998 same result

Query:
```sql
SELECT TOP 10 * FROM shopping_cart_conversion;
```

Result:
```bash
Expected "!=", "#", "%", "&", "&&", "\'", "(", "*", "+", ",", "-", "--", "->", "->>", ".", "/", "/*", ":=", ";", "<", "<<", "<=", "<>", "=", ">", ">=", ">>", "AND", "AS", "BETWEEN", "FOR", "FROM", "GO", "GROUP", "HAVING", "IN", "INTERSECT", "INTO", "IS", "LIKE", "LIMIT", "LOCK", "MINUS", "NOT", "OR", "ORDER", "REGEXP", "RLIKE", "UNION", "WHERE", "WINDOW", "XOR", "\\"", "^", "`", "div", "|", "||", "~", [ \\t\\n\\r], [A-Za-z_], or end of input but "1" found.
```

**Comment by taozhi8833998** [comment]
> ```sql
> SELECT TOP 10 * FROM shopping_cart_conversion;
> ```

it has been fixed, and will be released in the next version `4.10.0`

**Comment by taozhi8833998** [comment]
@kasvith [test case](https://github.com/taozhi8833998/node-sql-parser/blob/1d3c7f96b3e61b17b5386058cc7ffff4565dc659/test/transactsql.spec.js#L24C40-L24C40)
